packages/trpc/server/routers/viewer/workflows.tsx

```diff
@@ -49,32 +51,149 @@ function getSender(
   return isSMSAction(step.action) ? step.sender || SENDER_ID : step.senderName || SENDER_NAME;
 }
 
+async function isAuthorized(
+  workflow: Pick<Workflow, "id" | "teamId" | "userId"> | null,
+  prisma: PrismaClient<
+    Prisma.PrismaClientOptions,
+    never,
+    Prisma.RejectOnNotFound | Prisma.RejectPerOperation | undefined
+  >,
+  currentUserId: number
+) {
+  if (!workflow) {
+    return false;
+  }
+
+  const userWorkflow = await prisma.workflow.findFirst({
+    where: {
+      id: workflow.id,
+      OR: [
+        { userId: currentUserId },
+        {
+          team: {
+            members: {
+              some: {
+                userId: currentUserId,
```

Comment:
It doesn't check if the member has accepted the team request or not.
I am not sure how big of a problem it can be, but i think only those members should be able to read or edit a workflow which have accepted the team request.

Reply:
Thank you, didn't think about that! I added the check if the membership is accepted